tce/src/applibs/costdb/CostDatabaseRegistry.cc

```diff
@@ -48,10 +48,7 @@ CostDatabaseRegistry::CostDatabaseRegistry() {
  */
 CostDatabaseRegistry::~CostDatabaseRegistry() {
     AssocTools::deleteAllValues(registry_);
-    if (instance_ != NULL) {
```

Comment:
I don't get this fix, can you explain? Doesn't this leak the instance_ now?

Reply:
Please address this @kattervo ?

Reply:
Yeah I think that version gave segmentation fault during runtime, so Kati asked to do the modification.

Reply:
The previous version is malformed, that delete would lead into infinite recursion of `instance_` trying to delete itself, as `delete instance_` calls the destructor on that instance.

The test calls `delete` on that instance, so running the test doesn't leak memory, although the class can be used in a way where it does leak. I didn't manage to change it in a way that avoids a segfault/double free, but I didn't consider the destructor being called for `instance_` in the first place.

Does changing the if conditional to `instance_ != NULL && instance_ != this` avoid the segfault?

Reply:
Yes this indeed looks fishy - I wonder why it hasn't caused troubles before, we've been lucky. This is a singleton so I'd not try to delete at all (the process teardown should clear it up).  I'll pull this in.